### PR TITLE
Ben/extrinsics docs scraper

### PR DESCRIPTION
**Pull request by benluelo**
## Description
I wrote a scraper to extract extrinsics documentation. Documentation is written to `book/pallets/<pallet-name>/extrinsics.md` for every pallet that has a folder in the `book/pallets/` directory. Also works with mdbook and hot reloading; check out `make run-book` and `make build-book`.

## Checklist

This PR adds so much docs

- [x] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [x] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

**Comment by dzmitry-lahoda** [comment]
just question, afaik ts and rs clients are generated, do they copy docs?  or may be docs are copied into json rpc description? or may be we could build rust docs, and our md book just reference this? 

i was ok with generate rust socs and make mdbook reference this. 

and if generate, can we mark all generated docs as such?  to avoid manual editing? or generate during build of mdbook?

overall stuff ok, but given there are variants and tool is maintance burded, not sure.

**Comment by dzmitry-lahoda** [comment]
there are 2 way to prevent broken dockers, one way is reuse what wa done as @andor0 suggested. 

if not or few reuse, need to mark owners of that file in git to maintain, or it will be broken ,soon

**Comment by benluelo** [comment]
> just question, afaik ts and rs clients are generated, do they copy docs? or may be docs are copied into json rpc description? or may be we could build rust docs, and our md book just reference this?
> 
> i was ok with generate rust socs and make mdbook reference this.
> 
> and if generate, can we mark all generated docs as such? to avoid manual editing? or generate during build of mdbook?
> 
> overall stuff ok, but given there are variants and tool is maintance burded, not sure.

what do you mean by clients? this tool extracts the extrinsics docs from the pallets and puts it in the book (book/pallets/<pallet>/extrinsics.md)

Auto-generated comments are a good idea, added those :+1:

not sure what you mean about docker either?

File: frame/mosaic/src/lib.rs [hunk 1]
```diff
@@ -353,9 +353,14 @@ pub mod pallet {
...
-		/// inflight, incoming transactions from the previous relayer. Budgets remain in place
-		/// however.
+		/// Sets the current Relayer configuration.
+		///
+		/// This is enacted immediately and invalidates inflight/ incoming transactions from the
+		/// previous Relayer. However, existing budgets remain in place.
+		///
+		/// This can only be called by the [`ControlOrigin`].
+		///
+		/// [controlorigin]: https://dali.devnets.composablefinance.ninja/doc/pallet_mosaic/pallet/trait.Config.html#associatedtype.ControlOrigin
```
Comment (PoisonPhang) [inline frame/mosaic/src/lib.rs:363]: You can have relative links in cargo docs
See my example [here](https://github.com/ComposableFi/composable/blob/fd6649d63883cab4daf3d1ae238b749ba01232b0/frame/mosaic/src/lib.rs#L359)
And a more general note of the feature from the rust team [here](https://rust-lang.github.io/rfcs/1946-intra-rustdoc-links.html)

Comment (PoisonPhang) [reply frame/mosaic/src/lib.rs:363]: Though, I suppose this is better as if for when we aren't in cargo docs

**Review by PoisonPhang — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
why not just generate rust docs and put link from mdbook? 

clients - extinsic generate client for ts, may be npm package docs will have docs?

not sure this tool is needed.

**Comment by dzmitry-lahoda** [comment]
https://github.com/ComposableFi/composable/pull/874#issuecomment-1100766829

**Comment by dzmitry-lahoda** [comment]
https://github.com/ComposableFi/composable/pull/874#issuecomment-1100766829

**Comment by benluelo** [comment]
Closing in favor of #898 (I should have merged back into @cor's branch originally)

**Pull request closed without merging**